src/review/logic.py

```diff
@@ -50,28 +61,134 @@ def handle_review_file(_file, content_type, review_assignment, kind):
+        form=review_form
+    ).order_by('order')
+
+    for relation in relations:
+        if relation.element.field_type in ['text', 'textarea', 'date', 'email']:
+            document.add_heading(
+                relation.element.name + ": _______________________________",
+                level=1,
+            )
+            document.add_paragraph(relation.help_text).italic = True
+
+        if relation.element.field_type in ['select', 'check']:
+            document.add_heading(relation.element.name, level=1)
+
+            if relation.element.field_type == 'select':
+                choices = render_choices(relation.element.choices)
+            else:
+                choices = ['Y', 'N']
+
+            p = document.add_paragraph(relation.help_text)
+            p.add_run(
+                ' Mark your choice however you like, as long as it is clear.'
+            ).italic = True
+            table = document.add_table(rows=2, cols=len(choices))
+            hdr_cells = table.rows[0].cells
+
+            for i, choice in enumerate(choices):
+                hdr_cells[i].text = choice[0]
+
+            table.style = 'TableGrid'
+
+    document.add_page_break()
+
+    if not os.path.exists(os.path.join(settings.BASE_DIR, 'files', 'forms')):
+        os.makedirs(os.path.join(settings.BASE_DIR, 'files', 'forms'))
+
+    path = os.path.join(
+        settings.BASE_DIR,
+        'files',
+        'forms',
+        '%s.docx' % str(uuid4())
+    )
+
+    document.save(path)
+
+    return path
+
+
+@is_reviewer
+def serve_file(request, file_path):
+    try:
+        file_stream = default_storage.open(file_path, 'r')
+        mimetype = mime.guess_type(file_path)
+        response = StreamingHttpResponse(file_stream, content_type=mimetype)
+        response['Content-Disposition'] = (
+            "attachment; filename=review_form.docx"
         )
+        return response
+    except IOError:
+        messages.add_message(request, messages.ERROR, 'File not found.')
+        return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+
+
+def handle_editorial_review_file(
+        file,
+        proposal,
+        review_assignment,
+        kind,
+        editorial
+):
+    original_filename = smart_text(file._get_name())
+    filename = str(uuid4()) + str(os.path.splitext(original_filename)[1])
+    folder_structure = os.path.join(
+        settings.BASE_DIR,
+        'files',
+        'books',
+        str(review_assignment.book.id)
+    )
+
+    if not os.path.exists(folder_structure):
+        os.makedirs(folder_structure)
 
-    if not path.exists(folder_structure):
-        makedirs(folder_structure)
+    path = os.path.join(folder_structure, str(filename))
+    fd = default_storage.open(path, 'wb')
+
+    for chunk in file.chunks():
+        fd.write(chunk)
 
-    _path = path.join(folder_structure, str(filename))
-    fd = open(_path, 'wb')
-    [fd.write(chunk) for chunk in _file.chunks()]
     fd.close()
```

Comment:
This is also old code, but it may be worth changing it to use a `with` statement

Reply:
Thanks for spotting this. This is one of the functions I missed, and as it was, would have blown up or saved files in the wrong directory on s3. 

There's about 20 of these near-identical functions, and if time wasn't a factor, I'd refactor the shared logic into one function.

Reply:
I know exactly where you're coming from.